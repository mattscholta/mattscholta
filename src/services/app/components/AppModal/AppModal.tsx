import * as React from 'react';
import classnames from 'classnames';
import { useRecoilState } from 'recoil';

import { modalAtom } from '@src/recoil/atoms/modal';

import styles from './styles.module.css';

export interface AppModalProps {
  className?: string;
}

/**
 * @name AppModal
 * @description We interact with a singleton for our Modals which allows
 * us to keep the UI consistent and prevents us from having to deal with
 * potentially stacking issues.
 */
export const AppModal = (props: AppModalProps) => {
  const { className } = props;

  // Hooks
  const [state, _setState] = useRecoilState(modalAtom);

  // Setup
  const { open } = state;

  // Styles
  const cssComponent = classnames(styles.component, className, {
    [styles.close]: !open,
    [styles.open]: open
  });

  return (
    <div className={cssComponent}>
      <div className={styles.content}>
        <h2 className="u-mb-2x">App Modal</h2>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Laborum fugit
          nihil nesciunt quibusdam deleniti, adipisci, deserunt et aliquid
          aspernatur provident consequuntur, optio eaque ut debitis quo enim
          sequi dolores reiciendis?
        </p>
      </div>
    </div>
  );
};
