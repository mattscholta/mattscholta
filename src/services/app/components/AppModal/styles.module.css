.component {
  background-color: rgba(0 0 0 / 40%);
  bottom: 0;
  display: flex;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  transition: opacity 200ms ease-out;
  z-index: 10;
}

.component.close {
  opacity: 0;
  pointer-events: none;
}

.component.open {
  opacity: 1;
}

.content {
  background-color: theme('colors.white');
  height: 95vh;
  margin: auto;
  padding: theme('spacing.2x');
  transition: transform 300ms ease-in-out;
  width: 95vw;
  z-index: 1;
}

.close .content {
  transform: scale(1.3);
}

.open .content {
  pointer-events: all;
  transform: scale(1);
}

.underlay {
  background-color: rgba(0 0 0 / 40%);
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: 0;
}

.token {
  border: none;
  height: 350px;
  resize: none;
  width: 100%;
}

@screen lg {
  .content {
    height: 80vh;
    max-width: 1000px;
    padding: theme('spacing.4x');
    width: 95vw;
  }

  .close .content {
    transform: scale(1.3);
  }
}
