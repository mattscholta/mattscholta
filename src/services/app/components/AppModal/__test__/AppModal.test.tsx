import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { AppModal, AppModalProps } from '../AppModal';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '',
      push: jest.fn(),
      query: {},
      route: '/'
    };
  }
}));

describe('AppModal', () => {
  let props: AppModalProps;
  let wrapper: RenderResult;

  const createWrapper = () => (
    <RecoilRoot>
      <AppModal {...props} />
    </RecoilRoot>
  );
  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
