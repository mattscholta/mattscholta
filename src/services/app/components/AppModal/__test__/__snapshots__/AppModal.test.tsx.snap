// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal matches snapshot 1`] = `
NodeList [
  <div
    class="component close"
  >
    <div
      class="content"
    >
      <h2
        class="u-mb-2x"
      >
        App Modal
      </h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Laborum fugit nihil nesciunt quibusdam deleniti, adipisci, deserunt et aliquid aspernatur provident consequuntur, optio eaque ut debitis quo enim sequi dolores reiciendis?
      </p>
    </div>
  </div>,
]
`;
